src/wanderer/lib.ts

```diff
@@ -0,0 +1,248 @@
+   * @param prepareTurn attempt to set up, spending meat and or items as necessary
+   */
+  constructor(
+    name: string,
+    location: Location,
+    value: number,
+    prepareTurn: () => boolean = () => true
+  ) {
+    this.name = name;
+    this.value = value;
+    this.location = location;
+    this.prepareTurn = prepareTurn;
+  }
+}
+export type WandererFactory = (
+  type: DraggableFight,
+  locationSkiplist: Location[]
+) => WandererTarget[];
+export type WandererLocation = { location: Location; targets: WandererTarget[]; value: number };
+
+export const unsupportedChoices = new Map<Location, { [choice: number]: number | string }>([
+  [$location`The Spooky Forest`, { [502]: 2, [505]: 2 }],
+  [$location`Guano Junction`, { [1427]: 1 }],
+  [$location`The Hidden Apartment Building`, { [780]: 6, [1578]: 6 }],
+  [$location`The Black Forest`, { [923]: 1, [924]: 1 }],
+  [$location`LavaCo™ Lamp Factory`, { [1091]: 9 }],
+  [$location`The Haunted Laboratory`, { [884]: 6 }],
+  [$location`The Haunted Nursery`, { [885]: 6 }],
+  [$location`The Haunted Storage Room`, { [886]: 6 }],
+  [$location`The Hidden Park`, { [789]: 6 }],
+  [$location`A Mob of Zeppelin Protesters`, { [1432]: 1, [857]: 2 }],
+  [$location`A-Boo Peak`, { [1430]: 2 }],
+  [$location`Sloppy Seconds Diner`, { [919]: 6 }],
+  [$location`VYKEA`, { [1115]: 6 }],
+  [
+    $location`The Castle in the Clouds in the Sky (Basement)`,
+    {
+      [670]: 4,
+      [671]: 4,
+      [672]: 1,
+    },
+  ],
+  [
+    $location`The Haunted Bedroom`,
+    {
+      [876]: 1, // old leather wallet, 500 meat
+      [877]: 1, // old coin purse, 500 meat
+      [878]: 1, // 400-600 meat
+      [879]: 2, // grouchy spirit
+      [880]: 2, // a dumb 75 meat club
+    },
+  ],
+  [$location`The Copperhead Club`, { [855]: 4 }],
+  [$location`The Castle in the Clouds in the Sky (Top Floor)`, { [1431]: 1, [677]: 2 }],
+  [$location`The Hidden Office Building`, { [786]: 6 }],
+]);
+
+export function defaultFactory(): WandererTarget[] {
+  return [new WandererTarget("Default", $location`The Haunted Kitchen`, 0)];
+}
+
+export function maxBy<T>(array: T[], key: (t: T) => number): T {
+  return array
+    .map((t: T) => {
+      return { t, value: key(t) };
+    })
+    .reduce((prev, curr) => (prev.value < curr.value ? curr : prev)).t;
+}
+
+type WanderingSource = {
+  name: string;
+  item: Item;
+  max: number;
+  property: NumericProperty;
+  type: DraggableFight;
+};
+const WanderingSources: WanderingSource[] = [
+  {
+    name: "CMG",
+    item: $item`cursed magnifying glass`,
+    max: 3,
+    property: "_voidFreeFights",
+    type: "wanderer",
+  },
+  {
+    name: "Voter",
+    item: $item`"I Voted!" sticker`,
+    max: 3,
+    property: "_voteFreeFights",
+    type: "wanderer",
+  },
+  {
+    name: "Voter",
+    item: $item`"I Voted!" sticker`,
+    max: 3,
+    property: "_voteFreeFights",
+    type: "wanderer",
+  },
+  {
+    name: "Backup",
```

Comment:
We have a silght issue that we need to figure out:
we have a type of `wanderer` called a `wanderer`. That leaves me confused--I see this "backup" thing, and I immediately thing "but I thought that was a wanderer".

I don't have a great solution to this! but we should rename one of them. I think that is how we ended up with the admittedly inelegant 'DraggableFight' name